File: projects/09/java/TempTester.java
/* TempTester.java is a 'driver' to test class Temperature.
 * Student: ZeAi Sun
 * Date: Mar 17, 2023
 ************************************************************/

import java.util.Scanner;

public class TempTester  
{
  public static void main(String[] args)
  {
    // get all the necessary inputs
      Temperature baseTemp = Temperature.InputOp();
      Temperature limitTemp = Temperature.InputOp();
      
      Scanner keyboard = new Scanner(System.in);
      System.out.println("Enter the step value: ");
      Double stepValue = keyboard.nextDouble();
      keyboard.close();
     
      while (Temperature.lessThan(baseTemp, limitTemp)) {
        Temperature.OutputOp(Temperature.convertF(baseTemp));
        Temperature.OutputOp(Temperature.convertC(baseTemp));
        Temperature.OutputOp(Temperature.convertK(baseTemp));
        System.out.print("\n");
        baseTemp = Temperature.raise(baseTemp, stepValue);
      }

      System.out.println("\nAll tests passed!");

  }
}

File: projects/09/ruby/TempTester.rb
# TempTester.rb tests class Temperature and its operations
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: ZeAi Sun
# Date: Mar 18, 2023
####################################################

require 'test/unit/assertions'
include Test::Unit::Assertions

load 'Temperature.rb'

def testTemp 
   baseTemp = Temperature.InputOp
   limitTemp = Temperature.InputOp
   puts "Enter the step value: "
   stepValue = gets

   while Temperature.lessThan(baseTemp, limitTemp)
      Temperature.OutputOp(Temperature.convertF(baseTemp))
      Temperature.OutputOp(Temperature.convertC(baseTemp))
      Temperature.OutputOp(Temperature.convertK(baseTemp))
      puts "\n"
      baseTemp = Temperature.raise(baseTemp, stepValue)
   end

   puts "All tests passed!\n"
end

testTemp